compiler/rustc_typeck/src/hir_wf_check.rs

```diff
@@ -20,6 +20,7 @@ fn diagnostic_hir_wf_check<'tcx>(
     (predicate, loc): (ty::Predicate<'tcx>, WellFormedLoc),
 ) -> Option<ObligationCause<'tcx>> {
     let hir = tcx.hir();
+    let predicate = tcx.erase_regions(predicate);
```

Comment:
hmm... not sure how this fixes the ICE, as the invalid region will still be in the query arguments and thus cached. Or is the ICE on the return value's caching?

Reply:
I know other queries have schemes to avoid having inference vars in their arguments, maybe we can do the same thing for this query?

Reply:
We only call this query in one place, so we can just call `erase_regions` before invoking the query.

Reply:
We might start calling it in more places, right? Either way, I can move the erasure higher in the chain.

Reply:
I think we'll only ever want to call this from the place where we perform error handling.